subsys/net/ip/packet_socket.c

```diff
@@ -46,7 +48,13 @@ enum net_verdict net_packet_socket_input(struct net_pkt *pkt, uint8_t proto)
 	 * data part to be feed to non raw socket.
 	 */
 
+	orig_family = net_pkt_family(pkt);
+
 	net_pkt_set_family(pkt, AF_PACKET);
 
-	return net_conn_input(pkt, NULL, proto, NULL);
+	(void)net_conn_input(pkt, NULL, proto, NULL);
+
+	net_pkt_set_family(pkt, orig_family);
+
+	return NET_CONTINUE;
```

Comment:
What if `net_conn_input()` returns `NET_OK`? Isn't `pkt` already "consumed" and dereferenced inside `net_conn_input()`?

Reply:
Could be, I will investigate this. The packet socket is quite complex and convoluted, this needs to be re-written at some point. It is difficult to see how the system will behave just by looking the code, basically one needs to run a test to verify that it is working ok. I did some tests with echo-server and did not see immediate issues with this proposal, but can try again with more complex setup.

Reply:
In all other network stack places function returning NET_OK does consume the packet. So this place looks suspicious, as it doesn't respect that. At least we should make some assumption what **should** happen, so that if we run into another issue related to it, then at least we will know which part is misbehaving.

While testing is important, I don't think we should rely on that. We might solve one issue and introduce another one here. Also, it would be nice to cover the case that we are fixing now with tests, so that we don't reintroduce it when fixing something different.

Reply:
I found one corner case when testing with echo-server/client and when packet sockets were enabled.

I did not change the code to check NET_OK as it does not seem to ever happen for packet socket.